postgres/datadog_checks/postgres/statement_samples.py

```diff
@@ -358,29 +381,60 @@ def _run_explain(self, dbname, statement, obfuscated_statement):
                 return None
             return result[0][0]
 
+    @staticmethod
+    def _translate_setup_state_err_to_reason(state, err):
```

Comment:
I think this function is unnecessary. The construction of the event is done exclusively in `_collect_plan_for_statement` so I think it would be better to set the structure of the `code`/`reason` in that function. Also there is a lot of duplication that isn't really needed here. I would just remove this function altogether.